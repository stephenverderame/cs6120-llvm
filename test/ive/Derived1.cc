#include <cstdio>
int main()
{
    for (int I = 0; I < 10; ++I) {
        int J = I * 10;
        printf("%d\n", J);
    }
}