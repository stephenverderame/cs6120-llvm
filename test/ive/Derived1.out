Loop: 
1:                                                ; preds = %7, %0
  %.0 = phi i32 [ 0, %0 ], [ %8, %7 ]
  %2 = icmp slt i32 %.0, 10
  br i1 %2, label %4, label %3

Derived IV: %5 = .0 * 10
Basic IV:   %8 = add nsw i32 %.0, 1
Basic IV:   %.0 = phi i32 [ 0, %0 ], [ %8, %7 ]
